setuptools/tests/test_manifest.py

```diff
@@ -75,7 +76,7 @@ def touch(filename):
 )
 
 
-translate_specs = [
+translate_specs: List[Tuple[str, List[str], List[str]]] = [
```

Comment:
Shouldn't the typechecker be able to infer the value of this variable without the need for special annotation?

Reply:
Without the explicit definition:
mypy:
```
setuptools\tests\test_manifest.py:129: note: Revealed type is "builtins.list[Tuple[builtins.str, builtins.list[builtins.str], builtins.object]]"
setuptools\tests\test_manifest.py:143: error: No overload variant of "zip" matches argument types "repeat[str]", "object"  [call-overload]
setuptools\tests\test_manifest.py:143: note: Possible overload variants:
setuptools\tests\test_manifest.py:143: note:     def [_T_co] zip() -> zip[Any]
setuptools\tests\test_manifest.py:143: note:     def [_T_co, _T1] zip(Iterable[_T1], /) -> zip[Tuple[_T1]]
setuptools\tests\test_manifest.py:143: note:     def [_T_co, _T1, _T2] zip(Iterable[_T1], Iterable[_T2], /) -> zip[Tuple[_T1, _T2]]
setuptools\tests\test_manifest.py:143: note:     def [_T_co, _T1, _T2, _T3] zip(Iterable[_T1], Iterable[_T2], Iterable[_T3], /) -> zip[Tuple[_T1, _T2, _T3]]
setuptools\tests\test_manifest.py:143: note:     def [_T_co, _T1, _T2, _T3, _T4] zip(Iterable[_T1], Iterable[_T2], Iterable[_T3], Iterable[_T4], /) -> zip[Tuple[_T1, _T2, _T3, _T4]]
setuptools\tests\test_manifest.py:143: note:     def [_T_co, _T1, _T2, _T3, _T4, _T5] zip(Iterable[_T1], Iterable[_T2], Iterable[_T3], Iterable[_T4], Iterable[_T5], /) -> zip[Tuple[_T1, _T2, _T3, _T4, _T5]]
setuptools\tests\test_manifest.py:143: note:     def [_T_co] zip(Iterable[Any], Iterable[Any], Iterable[Any], Iterable[Any], Iterable[Any], Iterable[Any], /, *iterables: Iterable[Any]) -> zip[Tuple[Any, ...]]
```

pyright:
```
setuptools\setuptools\tests\test_manifest.py
  setuptools\setuptools\tests\test_manifest.py:129:13 - information: Type of "translate_specs" is "list[Unknown]"
```